awscli/customizations/autoprompt/history.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import os
+
+from prompt_toolkit.completion import Completion, Completer
+from prompt_toolkit.history import FileHistory
+
+from awscli.autocomplete.completer import CompletionResult
+from awscli.autocomplete.filters import fuzzy_filter
+
+
+class HistoryDriver(FileHistory):
```

Comment:
So I dived into this class a bit more. I'm thinking we just do our own implementation on top of the base `History` class. Mainly, I think I'd like to see us control our format/schema for recording commands. Specifically, I was thinking we just make this history file a json document (I can go either way on the order of commands):
```
{
    "version": "1",
    "commands": [
        "<oldest-command>"
        ...
        "<2nd-most-recent-command>"
        "<most-recent-command>",
     ]
}
```
I like this approach because:
1. We are in complete control of how completions are saved in the file. Right now our heuristic for saving max lines is based on what we expect prompt toolkit to be saving, but I'm not sure if there is any guarantee on how that is saved.
2. We can add a version string to the file so that we can disambiguate if we ever make a breaking change to the history format and need to account for different format versions.
3. We can load the contents of the history as a list making it a bit easier to interpret/work with.

Thoughts?